src/python/tests/test_maf_to_fasta.py

```diff
@@ -0,0 +1,133 @@
+
+    @pytest.mark.parametrize(
+        "genome_names, exp_output, expectation",
+        [
+            (['genomeA.1', 'genomeB.1'], re.compile('^(?P<genome>genomeA\\.1|genomeB\\.1)[.](?P<seqid>.+)$'),
+                does_not_raise()),
+            (iter(['genomeA.1', 'genomeB.1']), None, raises(ValueError)),
+            (['genomeA', 'genomeA.1'], None, raises(ValueError)),
+            ([], None, raises(ValueError))
+        ]
+    )
+    def test_compile_maf_src_regex(self, genome_names: Iterable[str], exp_output: Pattern[str],
+                                   expectation: ContextManager) -> None:
+        """Tests :func:`maf_to_fasta.compile_maf_src_regex()` function.
+
+        Args:
+            genome_names: The genome names expected to be in the input MAF file.
+            exp_output: Expected return value of the function.
+            expectation: Context manager for the expected exception. The test will only pass if that
+                exception is raised. Use :class:`~contextlib.nullcontext` if no exception is expected.
+
+        """
+        with expectation:
+            # pylint: disable-next=no-member
+            obs_output = maf_to_fasta.compile_maf_src_regex(genome_names)
+            assert obs_output == exp_output
+
+    @pytest.mark.parametrize(
+        "maf_file, output_dir, genomes_file, out_file_rel_paths, expectation",
+        [
+            ('gabs.maf', 'gabs', 'gab_genomes.txt', ['0.fa', '0.json', '1.fa', '1.json'], does_not_raise()),
+            ('gabs.maf', 'gabs', 'wrong_genomes.txt', [], raises(ValueError)),
+            ('gabs.maf', 'gabs', None, [], raises(ValueError))
+        ]
+    )
+    def test_convert_maf_to_fasta(self, maf_file: PathLike, output_dir: PathLike,
+                                  genomes_file: Optional[PathLike], out_file_rel_paths: List[str],
+                                  expectation: ContextManager, tmp_dir: Path) -> None:
+        """Tests :func:`maf_to_fasta.convert_maf_to_fasta()` function.
+
+        Args:
+            maf_file: Input MAF file with alignment blocks. The src fields of
+                this MAF file should be of the form '<genome>.<seqid>'.
+            output_dir: Output directory under which FASTA and JSON files will be created.
+            genomes_file: File listing the genomes in the input MAF file, one per line.
+            out_file_rel_paths: Relative paths of the output FASTA and JSON files.
+            expectation: Context manager for the expected exception. The test will only pass if that
+                exception is raised. Use :class:`~contextlib.nullcontext` if no exception is expected.
+            tmp_dir: Unit test temp directory (fixture).
+
+        """
+        with expectation:
+            maf_file_path = self.ref_file_dir / maf_file
+            out_dir_path = tmp_dir / output_dir
+            if genomes_file is not None:
+                genomes_file = self.ref_file_dir / genomes_file
```

Comment:
All the logic should be located **outside** the `with` block because, for instance, if you are expecting a `ValueError` and that is raised here (instead of in the actual call), the test will pass (but it shouldn't).